src/common/Formatter.h

```diff
@@ -54,6 +55,122 @@ namespace ceph {
       }
     };
 
+    /// Helper to check if a type is a map for our purpose
+    /// (based on fmt code)
+    template <typename T> class is_map {
```

Comment:
how about defining a concept? simpler this way.

Reply:
You have a point...
I just more or less copied the fmt code that makes this distinction.

Reply:
@tchaikov 
OK- after testing some options:
The main problem is one cannot define a concept within a class scope. And I do not wish to pollute the general scope.
A solution that was suggested is to add a static function. Something like:

```
    template<typename T>
    static constexpr auto my_is_map(T t)
    {
        return requires{ t.::mapped_type };
    }

```    
and change the usage into:
`if constexpr( my_is_map(t) )...
`
but I like that even less than SFINAE